pkg/cluster/ipaddresses.go

```diff
@@ -25,6 +25,16 @@ func (m *manager) updateRouterIP(ctx context.Context) error {
 	installConfig := g[reflect.TypeOf(&installconfig.InstallConfig{})].(*installconfig.InstallConfig)
 	kubeadminPassword := g[reflect.TypeOf(&password.KubeadminPassword{})].(*password.KubeadminPassword)
 
+	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		doc.OpenShiftCluster.Properties.APIServerProfile.URL = "https://api." + installConfig.Config.ObjectMeta.Name + "." + installConfig.Config.BaseDomain + ":6443/"
+		doc.OpenShiftCluster.Properties.ConsoleProfile.URL = "https://console-openshift-console.apps." + installConfig.Config.ObjectMeta.Name + "." + installConfig.Config.BaseDomain + "/"
+		doc.OpenShiftCluster.Properties.KubeadminPassword = api.SecureString(kubeadminPassword.Password)
+		return nil
+	})
+	return err
+}
+
+func (m *manager) updateRouterIP(ctx context.Context) error {
```

Comment:
1. the following code is appropriate in the context of install, but is it appropriate that it should cause an error in adminupgrade?

```golang
	if len(svc.Status.LoadBalancer.Ingress) == 0 {
		return fmt.Errorf("routerIP not found")
	}
```

2. In admin upgrade I would prefer to see us test the router IP address to see if it needs changing, and only change it if it really does.  It saves a gratuitous API write call per cluster.

Reply:
3. Also, what happens if there's no router-default service?

Reply:
> test the router IP address to see if it needs changing
Amber suggests that code might want to live in the DNS manager.